### ISSUE TITLE
Spring Web @RequestHeader(Map<String,String>) throws RuntimeException

### ISSUE THREAD

**Issue by jclingan** [quarkusio/quarkus#14051]
**Describe the bug**
I'm trying to access all request headers using Spring `@RequestHeader` in the same way that `@Context HttpHeaders headers` would provide access to all request headers.

When starting the application with `mvn quarkus:dev`, I get the following exception
>Caused by: java.lang.RuntimeException: RESTEASY003875: Unable to find a constructor that takes a String param or a 
>valueOf() or fromString() method for javax.ws.rs.HeaderParam("headers") on public java.lang.String 
>org.acme.SpringGreetingController.hello(java.util.Map) for basetype: java.util.Map

**Expected behavior**
A populated Map with HTTP headers, preferably a MultiValuedMap.

**Actual behavior**
Above exception.

**To Reproduce**

See [reproducer](https://github.com/jclingan/quarkus-spring-request-headers-error). Note, I'd like to use the Spring `MultiValueMap`, although the reproducer uses a `Map`.
The reproducer is literally a project generated from code.quarkus.io with the Spring Web extension, with the `hello()` method updated as follows;
```
    public String hello(@RequestHeader MultiValueMap<String, String> headers) {
        headers.forEach((key, value) -> {
            System.out.println(key + "->" + value);
        })
```

**Comment by geoand** [comment]
Thanks for reporting @jclingan! 

This will be a whole a lot easier to fix when we pivot the spring web compatibility to RESTEasy Reactive, so unless you really really need this soon, I'd rather wait.

**Comment by jclingan** [comment]
This is for the [book](https://www.manning.com/books/kubernetes-native-microservices-with-quarkus-and-microprofile) Ken and I are writing. Yeah, shameless plug, LOL. I can put a NOTE in the book for now and wait for the fix.

**Comment by geoand** [comment]
Great, thanks 👍

**Comment by jclingan** [comment]
Any update?

**Comment by geoand** [comment]
Not yet.

I hope to rebase the Spring Web support on RESTEasy Reactive some time before 2.x

**Comment by geoand** [comment]
I will likely work on this after the 2.2 stabilization release is out

**Comment by geoand** [comment]
I had to revert the "rebase" onto RESTEasy Reactive.
I will hopefully get back to it before 2.5

**Comment by liangyuanpeng** [comment]
@geoand  Thanks for your work and how can we get the request headers if we can't use `@RequestHeader` 

I have try this and also failed. It just tell me `javaax.ws.rs.NorSupportException`
```
 @GetMapping
  public String hello(@Context HttpHeaders headers) {
```

Thanks any help.

**Comment by geoand** [comment]
@liangyuanpeng yup, as you mention one can mix and match the Spring and JAX-RS annotations and types.

**Comment by aureamunoz** [comment]
Hi! is this issue still relevant? @jclingan @liangyuanpeng  If yes, can I have a reproducer? Otherwise, can I close it?

**Comment by aureamunoz** [comment]
cc @aureamunoz

**Comment by aureamunoz** [comment]
This morning I saw this in the lottery report and felt motivated to take a look: https://github.com/quarkusio/quarkus/pull/56253
